mlflow/java/src/main/java/org/mlflow/sagemaker/MLeapPredictor.java

```diff
@@ -1,18 +1,109 @@
+    predictionColumnNames =
+        JavaConverters.asScalaIteratorConverter(predictionColumnList.iterator()).asScala().toSeq();
+  }
+
+  /**
+   * Constructs an {@link MLeapPredictor}
+   *
+   * @param modelDataPath The path to the serialized MLeap model
+   * @param inputSchema The path to JSON-formatted file containing the input schema that the model
+   *     accepts
+   */
+  public MLeapPredictor(String modelDataPath, String inputSchemaPath) {
+    MleapContext mleapContext = new ContextBuilder().createMleapContext();
+    BundleBuilder bundleBuilder = new BundleBuilder();
+    this.pipelineTransformer = bundleBuilder.load(new File(modelDataPath), mleapContext).root();
+    try {
+      this.inputSchema = LeapFrameSchema.fromPath(inputSchemaPath);
+    } catch (IOException e) {
+      logger.error("Could not read the model input schema from the specified path", e);
+      throw new PredictorLoadingException(String.format(
+          "Failed to load model input schema from specified path: %s", inputSchemaPath));
+    }
+  }
 
   @Override
-  protected PredictorDataWrapper predict(PredictorDataWrapper input) {
-    throw new UnsupportedOperationException("Not yet implemented!");
+  protected PredictorDataWrapper predict(PredictorDataWrapper input)
+      throws PredictorEvaluationException {
+    PandasRecordOrientedDataFrame pandasFrame = null;
+    try {
+      pandasFrame = PandasRecordOrientedDataFrame.fromJson(input.toJson());
+    } catch (IOException e) {
+      logger.error(
+          "Encountered a JSON conversion error during conversion of Pandas dataframe to LeapFrame.",
+          e);
+      throw new PredictorEvaluationException(
+          "Failed to transform input into a JSON representation of an MLeap dataframe."
+          + "Please ensure that the input is a JSON-serialized Pandas Dataframe"
+          + "with the `record` orientation");
+    }
+
+    DefaultLeapFrame leapFrame = null;
+    try {
+      leapFrame = pandasFrame.toLeapFrame(this.inputSchema);
+    } catch (InvalidSchemaException e) {
+      throw new PredictorEvaluationException(
+          "Encountered a schema mismatch when converting the input dataframe to a LeapFrame.");
+    } catch (Exception e) {
+      logger.error(
+          "Encountered an unknown error during conversion of Pandas dataframe to LeapFrame.", e);
+      throw new PredictorEvaluationException(
+          "An unknown error occurred while converting the input dataframe to a LeapFrame.");
+    }
+
+    DefaultLeapFrame predictionsFrame =
+        this.pipelineTransformer.transform(leapFrame).get().select(predictionColumnNames).get();
+    Seq<Row> predictionRows = predictionsFrame.collect();
+    Iterable<Row> predictionRowsIterable =
+        JavaConverters.asJavaIterableConverter(predictionRows).asJava();
+    List<Object> predictions = new ArrayList<Object>();
+    for (Row row : predictionRowsIterable) {
+      predictions.add(row.getRaw(0));
```

Comment:
Why `getRaw(0)`? What if there are multiple predictionColumnNames?

Reply:
We only want to use rows from the column named `prediction` - this is what our `pyfunc` wrapper for Spark models does. Accordingly, the statically definitely `predictionColumnNames` list is misleading; I defined it because the `LeapFrame.select()` operation takes a list. I think it would be better to statically define the column name as a string and create a new list containing the single column name within the method that calls `LeapFrame.select()`.

After performing a select on a single-element list of column names, we can then select the first and only element of each row via `getRaw(0)`.